src/hooks/useToggle.ts

```diff
@@ -1,17 +1,22 @@
-import {useState} from "react";
+import { useState } from "react";
 
-export type UseToggle = [boolean, (value?: boolean) => void];
+export type UseToggle = [boolean, (value?: boolean) => void, () => void, () => void];
 
 export function useToggle(initialValue: boolean = false): UseToggle {
 	const [state, setState] = useState<boolean>(initialValue);
 	const toggleState = (value?: boolean) => {
 		if (value === undefined) {
-			setState(!state);
+			setState((v) => !v);
 		} else {
 			setState(value);
 		}
 	};
-	return [state, toggleState];
+
+	const turnOn = () => setState(true);
+
+	const turnOff = () => setState(false);
+
+	return [state, toggleState, turnOn, turnOff];
```

Comment:
I always prefer brevity, so I'd make these even easier readable:

```suggestion

	const on = () => setState(true);

	const off = () => setState(false);

	return [state, toggleState, on, off];
```